CONTRIBUTING.md

```diff
@@ -92,12 +106,121 @@ Commit messages are linted on commit, so you'll know if your message is not quit
+
+```bash
+git clone https://github.com/YOUR-USERNAME/vendure.git
+cd vendure
+```
+
+Add the [Vendure repository](https://github.com/vendure-ecommerce/vendure) as the upstream remote to your clone:
+
+```bash
+git remote add upstream https://github.com/vendure-ecommerce/vendure.git
+```
+
+This lets you pull updates from the original repository directly to your local clone.
+
+### Staying up to date
+> [!TIP]
+> It is a good idea to regularly update your local and forked repositories. You can do so by running the following commands:
+> ```bash
+> git fetch upstream
+> git checkout master
+> ```
+> Now you're on the master branch of your local repository, and Git "knows" whether there are differences between your master branch and the upstream branch.
+>
+> To update your local and forked remote repositories: 
+> ```bash
+> git merge upstream/master
+> git push origin master
+> ```
+
+### Create a new branch
+
+This is where you can start implementing your changes:
+
+```bash
+git checkout -b your-new-branch
+```
+
+### Developing
+
+Follow our [development guide](https://github.com/vendure-ecommerce/vendure#development) to make sure you have a properly set up development environment.
+
+After implementing your changes, stage the changes and commit them. Refer to the [Commit message format](#commit-message-format)
+```
+git add .
+git commit -m "type(scope): Message in present tense"
+```
+
+### Creating a pull request
+
+> [!TIP]
+> Now would be a good time to [update](#staying-up-to-date) your local and remote repositories.
 
-## Contributing Admin UI translations
+Make sure you have your new branch checked out and rebase it onto your up-to-date master branch:
+```bash
+git checkout your-new-branch
+git rebase master
+```
+Now push your changes to your forked repository:
+```bash
+git push origin your-new-branch
+```
+
+You can now create a pull request. If you prefer using the [GitHub CLI](https://cli.github.com/), run the following command:
+```bash
+gh pr create
+```
+and follow the prompts, this will automatically use our [pull request template](https://github.com/vendure-ecommerce/vendure/blob/master/.github/pull_request_template.md).
+
+Alternatively, navigate to your forked repository on GitHub and create a pull request from there.
+
+Well done! Now comes our part: we will review your pull request and either merge it or provide you with feedback on what we would like to see changed.
+
+## Contributing to the documentation
+
+For our [documentation](https://docs.vendure.io/), we use [Docusaurus](https://docusaurus.io/).
+
+After [setting up your development environment](#setting-up-the-dev-environment). Navigate to the docs directory and start the docusaurus server:
+```bash
+cd docs
+npm run start
+```
+
+this will output a URL where you can preview your changes.
+
+You can find all of our documentation built by docusaurus in these directories: 
+
+```
+docs/docs
+├── guides
+├── reference
+└── user-guide
+```
```

Comment:
Worth noting here that any docs in the "reference" dir are generated and should not be edited. Instead, they need to find the corresponding source file and edit the doc block, and the run the doc Generation script from the root package.json